src/core/thread/mesh_forwarder.cpp

```diff
@@ -421,6 +417,22 @@ ThreadError MeshForwarder::SendMessage(Message &aMessage)
     case Message::kTypeMacDataPoll:
         aMessage.SetDirectTransmission();
         break;
+
+    case Message::kTypeSupervision:
+        child = mNetif.GetChildSupervisor().GetDestination(aMessage);
+        VerifyOrExit(child != NULL, error = kThreadError_Drop);
+
+        if (child->IsRxOnWhenIdle())
+        {
+            aMessage.SetDirectTransmission();
+        }
```

Comment:
remove this `if` condition?
supervision message is only for sleepy child, that is, indirect message.